File: WriteFile/app/src/main/java/dk/it/writefile/ItemsDB.java
package dk.it.writefile;

import android.content.Context;

import java.util.HashMap;

public class ItemsDB {
    private static ItemsDB sItemsDB;
    private final HashMap<String, String> itemsMap = new HashMap<String, String>();

    private ItemsDB() {
        fillItemsDB();
    }

    public static ItemsDB get(Context context) {
        if (sItemsDB == null) sItemsDB = new ItemsDB();
        return sItemsDB;
    }

    public String listItems() {
        String r = "";
        for (HashMap.Entry<String, String> item : itemsMap.entrySet())
            r = r + "\nBuy " + item.getKey() + " in: " + item.getValue();
        return r;
    }

    private void fillItemsDB() {
        itemsMap.put("coffee", "Irma");
        itemsMap.put("carrots", "Netto");
        itemsMap.put("milk", "Netto");
        itemsMap.put("bread", "bakery");
        itemsMap.put("butter", "Irma");
    }
}

File: ShoppingXML/app/src/main/java/dk/itu/shopping/ItemsDB.kt
package dk.itu.shopping

class ItemsDB {
    val items = ArrayList<Item>()

    init {
        fillItemsDB()
    }

    companion object {
        private var sItemsDB: ItemsDB? = null

        fun get(): ItemsDB {
            return sItemsDB ?: ItemsDB().also { sItemsDB = it }
        }
    }

    fun addItem(what: String, where: String) {
        items.add(Item(what, where))
    }

    fun removeItem(what: String, where: String) {
        items.removeIf { it.what == what && it.where == where }
    }

    fun size(): Int {
        return items.size
    }

    private fun fillItemsDB() {
        items.add(Item("coffee", "Irma"))
        items.add(Item("carrots", "Netto"))
        items.add(Item("milk", "Netto"))
        items.add(Item("bread", "bakery"))
        items.add(Item("butter", "Irma"))
        items.add(Item("apples", "Netto"))
        items.add(Item("oranges", "Netto"))
        items.add(Item("cheese", "Netto"))
        items.add(Item("juice", "Føtex"))
        items.add(Item("onions", "Netto"))
        items.add(Item("potatoes", "Menu"))
        items.add(Item("chips", "Netto"))
        items.add(Item("coke", "Liedl"))
        items.add(Item("soap", "Netto"))
        items.add(Item("cookies", "Irma"))
        items.add(Item("ham", "Irma"))
    }
}